fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1721,23 +1728,31 @@ def _post_backward_hook(self, param: Parameter, *unused: Any) -> None:
         # reductions in post_backward stream.
         self._streams["post_backward"].wait_stream(torch.cuda.current_stream())
         with torch.cuda.stream(self._streams["post_backward"]):
-            orig_grad_data = param.grad.data
 
             if self.fp32_reduce_scatter:
                 # Cast grad to FP32.
                 param.grad.data = param.grad.data.float()
 
+            orig_grad_data = param.grad.data
```

Comment:
Move here to make `orig_grad_data` FP32. This was from https://github.com/facebookresearch/fairscale/pull/1139#discussion_r1342204098